### PR TITLE
[secrets/ctx] Modify context to include secrets directly

### PR DESCRIPTION
**Pull request by tenczar**
This change places the contents of secrets directly into the secrets map within the context object. This allows end users to directly access the secret values by their key without needing to specify the name of the secret define within dispatch.

### CODE REVIEW

File: pkg/function-manager/handlers.go [hunk 1]
```diff
@@ -123,9 +123,12 @@ func functionModelOntoEntity(m *models.Function, e *functions.Function) error {
 func runModelToEntity(m *models.Run, f *functions.Function) *functions.FnRun {
 	defer trace.Trace("runModelToEntity")()
 	secrets := f.Secrets
-	if m.Secrets != nil && len(m.Secrets) > 0 {
+	if secrets == nil {
 		secrets = m.Secrets
+	} else {
+		secrets = append(secrets, m.Secrets...)
```
Comment (kars7e) [inline pkg/function-manager/handlers.go]: I was debating with myself whether we should append or replace here. I understand that if function secrets do not overlap with secrets passed to a function run, all of them will be available to the function. If there is an overlap (two secrets with the same key inside), the secret defined in the function run will take precedence, as it has higher slice index and thus will overwrite function default secret. 

If my understanding above is correct, this behavior seems ok.

Comment (berndtj) [reply pkg/function-manager/handlers.go]: I think your understanding is correct.  Could we add a comment explaining what the intended behavior is?  It's not immediately obvious that the order matters.

File: pkg/functions/secretinjector/secretinjector.go [hunk 2]
```diff
@@ -55,7 +55,11 @@ func (injector *secretInjector) getSecrets(secretNames []string, cookie string)
 
 			return secrets, err
 		}
-		secrets[*resp.Payload.Name] = resp.Payload.Secrets
+
+		for key, value := range resp.Payload.Secrets {
+			secrets[key] = value
+		}
+		//secrets[*resp.Payload.Name] = resp.Payload.Secrets
```
[the remark this answers is not in the document]
Comment (berndtj) [reply pkg/functions/secretinjector/secretinjector.go]: This isn't a true "merge" of secrets.  I'm not sure if this is OK or not.  I guess we can go with the simple approach and find out...  For example:
```
secret1 := `{"key1": {"subkey1": "subvalue1"}}`
secret2 := `{"key1": {"subkey2": "subvalue2"}}`
```
The merged value via the simple algorithm is just secret2, but via a true merge would be:

```
merged := `{"key1": {"subkey1": "subvalue1", "subkey2": "subvalue2"}}`
```

Any comments?

[the remark this answers is not in the document]
Comment (kars7e) [reply pkg/functions/secretinjector/secretinjector.go]: My understanding was that `key` in:
```
for key, value := range resp.Payload.Secrets {
     secrets[key] = value
}
```
Is the subkey from your example, not secret1 or secret2 (those are secret names, iterated over `secretNames`). If that's the case, then merge will look like your example `merged` map.

[the remark this answers is not in the document]
Comment (berndtj) [reply pkg/functions/secretinjector/secretinjector.go]: Oh, I think you're right with regards to my sample.  I guess I would have to go one level deeper, but the point is the same.  I'm not going to worry about it now.  It's just a question of whether merge just overwrites values, or if it merges values.

[the remark this answers is not in the document]
Comment (tenczar) [reply pkg/functions/secretinjector/secretinjector.go]: Hmmmm... so the interesting thing about a secret is that it currently looks like this 
```
{
    "name" : "open-sesame",
    "secrets" : {
          "key1" : value1,
          "key2" : value2
    }
}
```
where value1 and value2 can be anything, i.e. any primitive or object (hashmap).
In the context object we currently retrieve secrets this way ```context.secrets["open-sesame"].key1```. The goal of this change was to remove the dependence on the secret name. So now you would do ```context.secrets["key1"]```. Do we really want to start making assumptions about the contents of the keys?

For example, what is the expected output for this:
Secret 1:
```
{
    "name" : "open-sesame",
    "secrets" : {
        "password" : "ali-baba"
    }
}

Secret 2:
```
{
    "name"  : "open-sesame",
    "secrets" : {
        "password" : {"plain" : "Aladdin", "base64" : "19ba3df21eac3"}
    }
}

Context Obj:
```
{
    ...,
    "secrets" : {
        "password" :  ?
    }
}

Keep in mind, removing the dependence on the name of the secret was the point of this PR.

[the remark this answers is not in the document]
Comment (kars7e) [reply pkg/functions/secretinjector/secretinjector.go]: I agree with @tenczar here, there is no natural way to merge those, I don't see how that could work without inspecting the value, which should be out of our concern.

[the remark this answers is not in the document]
Comment (berndtj) [reply pkg/functions/secretinjector/secretinjector.go]: I'm ok with this

**Pull request merged**